library/src/main/java/net/ossrs/yasea/SrsFlvMuxer.java

```diff
@@ -641,25 +638,28 @@ public void muxSequenceHeader(ByteBuffer sps, ByteBuffer pps,
         }
 
         private int searchAnnexb(ByteBuffer bb) {
-            for (int offset = bb.position(); offset < bb.capacity(); offset++) {
-                int size = bb.capacity() - offset;
+            while (bb.position() < bb.capacity()) {
+                int offset = bb.position();
+                int size = bb.remaining();
 
                 // match N[00] 00 00 00 01, where N>=0
                 if (size > 3 && bb.get(offset) == 0x00 && bb.get(offset + 1) == 0x00 && bb.get(offset + 2) == 0x00 && bb.get(offset + 3) == 0x01) {
-                    return offset + 4;
+                    return 4;
```

Comment:
Minor improvement: Hardcoding return values or conditions makes it difficult for code analysis.